File: wrappers/javascript/shared/src/api/CredentialDefinitionPrivate.ts
import { AnoncredsObject } from '../AnoncredsObject'
import { anoncreds } from '../register'

export class CredentialDefinitionPrivate extends AnoncredsObject {
  public static load(json: string) {
    return new CredentialDefinitionPrivate(anoncreds.credentialDefinitionPrivateFromJson({ json }).handle)
  }
}

File: wrappers/kotlin/src/commonMain/kotlin/api/CredentialDefinitionPrivate.kt
package anoncreds.api

import anoncreds_rs.ObjectHandle
import anoncreds_rs.anoncreds_credential_definition_private_from_json

class CredentialDefinitionPrivate
internal constructor(
    handle: ObjectHandle
) : AnoncredsObject(handle) {

    companion object{
        fun fromJson(json: String): CredentialDefinitionPrivate {
            return fromJson(
                json,
                ::CredentialDefinitionPrivate,
                ::anoncreds_credential_definition_private_from_json
            )
        }
    }
}